pkg/jobs/jobs.go

```diff
@@ -0,0 +1,128 @@
+package jobs
+
+import (
+	"bytes"
+	"encoding/gob"
+	"encoding/json"
+	"fmt"
+	"math/rand"
+	"time"
+)
+
+const (
+	// Queued state
+	Queued State = "queued"
+	// Running state
+	Running = "running"
+	// Errored state
+	Errored = "errored"
+)
+
+const (
+	// JSONEncoding is a JSON encoding message type
+	JSONEncoding = "json"
+	// GobEncoding is a Gob encoding message type
+	GobEncoding = "gob"
+)
+
+type (
+	// Queue interface is used to represent a asynchronous queue of jobs from
+	// which it is possible to enqueue and consume jobs.
+	Queue interface {
+		Enqueue(*Job) error
+		Consume() (bool, *Job, error)
+		Foreach(func(*Job) error) error
+		Len() int
+		Close()
+	}
+
+	// Broker interface is used to represent a job broker associated to a
+	// particular domain from which it is possible to create jobs that a pushed
+	// in the job system.
```

Comment:
No. However for the case of the memory queue it is the case, but it is particular.

I could add an explicit interface representing the responsibility to start the workers... but I'm not sure how to do it for now.